code/game/objects/items/devices/radio/electropack.dm

```diff
@@ -143,4 +143,82 @@ Code:
 </TT>"}
 	user << browse(dat, "window=radio")
 	onclose(user, "radio")
-	return
\ No newline at end of file
+	return
+
+/obj/item/device/electropack/shockcollar
+	name = "shock collar"
+	desc = "A reinforced metal collar. It seems to have some form of wiring near the front. Strange.."
+	icon = 'icons/obj/clothing/neck.dmi'
+	icon_state = "shockcollar"
+	item_state = "shockcollar"
+	body_parts_covered = NECK
+	slot_flags = SLOT_NECK
+	w_class = WEIGHT_CLASS_SMALL
+	strip_delay = 60
+	equip_delay_other = 60
+	materials = list(MAT_METAL=5000, MAT_GLASS=2000)
+	var/tagname = null
+
+/obj/item/device/electropack/shockcollar/attack_hand(mob/user)
+	if(loc == user)
+		if(slot_flags == SLOT_NECK)
+			to_chat(user, "<span class='warning'>The collar is fastened tight! You'll need help taking this off!</span>")
+			return
+	..()
+
+/obj/item/device/electropack/shockcollar/receive_signal(datum/signal/signal)
+	if(!signal || signal.encryption != code)
+		return
+
+	if(isliving(loc) && on)
+		if(shock_cooldown != 0)
+			return
+		shock_cooldown = 1
+		spawn(100)
```

Comment:
either use addtimer here, or make shock_cooldown use world.time and check it against world.time instead.

Reply:
Alright. Will note this is what the original electropack does. So will tweak for it too. Electropack code is ooooold.